ci-operator/step-registry/baremetalds/devscripts/conf/ipstack/baremetalds-devscripts-conf-ipstack-ref.yaml

```diff
@@ -0,0 +1,14 @@
+ref:
+  as: baremetalds-devscripts-conf-ipstack
```

Comment:
Thanks for the PR. IP_STACK is gotten from jobs as environment variable and all variants have been defined as an example seen here https://github.com/openshift/release/blob/ee576edc4a4c33496cea00c2566c34a3e3c569a3/ci-operator/config/openshift/cluster-baremetal-operator/openshift-cluster-baremetal-operator-release-4.9.yaml#L107

Additionally, we are trying to deprecate devscripts-conf and put more effort using pre and post chains like this https://github.com/openshift/release/blob/master/ci-operator/step-registry/baremetalds/e2e/serial/ipv4/baremetalds-e2e-serial-ipv4-workflow.yaml

Reply:
The reasoning for this step to configure outside of the `DEVSCRIPTS_CONFIG` is because from a `DEVSCRIPTS_CONFIG` perspective the difference between our two workflows is only the `IP_STACK` variable. If possible I would like to avoid creating two workflows for only a change in one variable. If I need to override in the test without this step it requires me to copy most of the same config to change just this variable as well.

For connected flows the `DEVSCRIPTS_CONFIG` is:
IP_STACK=v4
NUM_EXTRA_WORKERS=1
EXTRA_WORKER_VCPU=8
EXTRA_WORKER_MEMORY=32768
EXTRA_WORKER_DISK=120
PROVISIONING_NETWORK_PROFILE=Disabled
REDFISH_EMULATOR_IGNORE_BOOT_DEVICE=True

and for disconnected flows it is the following:
IP_STACK=v6
NUM_EXTRA_WORKERS=1
EXTRA_WORKER_VCPU=8
EXTRA_WORKER_MEMORY=32768
EXTRA_WORKER_DISK=120
PROVISIONING_NETWORK_PROFILE=Disabled
REDFISH_EMULATOR_IGNORE_BOOT_DEVICE=True

Most of these env variables are set in the -ztp-worflow I created aside from the IP_STACK variable so it is very similar to the one you linked.

Reply:
We had same discussion about what is the best way to manage for example if there is only one variable need to be changed. Actually best practice is;
NUM_EXTRA_WORKERS=1
EXTRA_WORKER_VCPU=8
EXTRA_WORKER_MEMORY=32768
EXTRA_WORKER_DISK=120
PROVISIONING_NETWORK_PROFILE=Disabled
REDFISH_EMULATOR_IGNORE_BOOT_DEVICE=True

in one chain and

IP_STACK=v4 is in another chain.

However, this is not supported(environment variable merging). That's why, we conclude that creating two different workflows and copying everything. I highly suggest that. But if you decide not move that way, I'd prefer changing the name `ipstack` something to `assisted` or etc. Because this new `devscripts-conf` is only used for assisted workflows.